File: components/tabs/flutter/ciata_tabs.dart
import 'package:flutter/material.dart';

/// Implementação experimental do CMP-0013 Tabs para Flutter.
class CiataTabs extends StatelessWidget {
  const CiataTabs({
    super.key,
    required this.labels,
    required this.selectedIndex,
    required this.onSelectionChange,
    required this.panelBuilder,
  })  : assert(labels.length > 0, 'labels não pode ser vazio'),
        assert(selectedIndex >= 0 && selectedIndex < labels.length,
            'selectedIndex fora do intervalo válido');

  final List<String> labels;
  final int selectedIndex;
  final ValueChanged<int> onSelectionChange;
  final Widget Function(int) panelBuilder;

  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      length: labels.length,
      initialIndex: selectedIndex,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          TabBar(
            onTap: onSelectionChange,
            tabs: [for (final label in labels) Tab(text: label)],
          ),
          panelBuilder(selectedIndex),
        ],
      ),
    );
  }
}

File: components/tabs/android/CiataTabs.kt
package org.ciata.ds.tabs

import androidx.compose.foundation.layout.Column
import androidx.compose.material3.Tab
import androidx.compose.material3.TabRow
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

/** Implementação experimental do CMP-0013 Tabs para Jetpack Compose. */
@Composable
fun CiataTabs(
    labels: List<String>,
    selectedIndex: Int,
    onSelectionChange: (Int) -> Unit,
    content: @Composable (Int) -> Unit,
) {
    val normalizedLabels = labels.map(String::trim)

    require(normalizedLabels.isNotEmpty()) { "labels não pode ser vazio." }
    require(selectedIndex in normalizedLabels.indices) { "selectedIndex fora do intervalo válido." }
    require(normalizedLabels.all { it.isNotEmpty() }) { "rótulos de abas não podem ser vazios." }

    Column {
        TabRow(selectedTabIndex = selectedIndex) {
            normalizedLabels.forEachIndexed { index, label ->
                Tab(
                    selected = index == selectedIndex,
                    onClick = { onSelectionChange(index) },
                    text = { Text(label) },
                )
            }
        }
        content(selectedIndex)
    }
}
